### PR TITLE
tetra: use field filters when reading via `io_reader_client`

### PR DESCRIPTION
**Pull request by mtardy**
Previously the field filters were used only in the server-side code:
https://github.com/cilium/tetragon/blob/cd96d7ca3d7556f0d44da7da0ab4cdf943d29957/pkg/server/server.go#L145-L149

### CODE REVIEW

**Review by willfindlay — APPROVED** [review]
Thanks!

**Comment by willfindlay** [comment]
@mtardy is this ready to merge or did you want to add some tests first?

**Comment by mtardy** [comment]
> @mtardy is this ready to merge or did you want to add some tests first?

It's ready I added them already! https://github.com/cilium/tetragon/pull/668/commits/628f1cee7c08c8092f94837808337d51fd142993

File: cmd/tetra/getevents/getevents_test.go [hunk 1]
```diff
@@ -64,3 +66,43 @@ func Test_GetEvents_Process(t *testing.T) {
 		assert.Equal(t, 2, bytes.Count(output, []byte("\n")))
 	})
 }
+
+func Test_GetEvents_FilterFields(t *testing.T) {
```
Comment (kkourt) [inline cmd/tetra/getevents/getevents_test.go:70]: Thanks for the test!

**Review by kkourt — APPROVED** [review]
🚀

**Pull request merged**